homeassistant/components/upnp/__init__.py

```diff
@@ -222,13 +223,29 @@ class UpnpEntity(CoordinatorEntity):
 
     coordinator: UpnpDataUpdateCoordinator
 
-    def __init__(self, coordinator: UpnpDataUpdateCoordinator) -> None:
+    def __init__(
+        self,
+        coordinator: UpnpDataUpdateCoordinator,
+        sensor_entity: EntityDescription,
+    ) -> None:
         """Initialize the base entities."""
         super().__init__(coordinator)
         self._device = coordinator.device
+        self._key = sensor_entity.key
+        self._attr_name = f"{coordinator.device.name} {sensor_entity.name}"
+        self._attr_unique_id = f"{coordinator.device.udn}_{sensor_entity.key}"
+        self._attr_icon = sensor_entity.icon
```

Comment:
```suggestion
```
Not needed

Reply:
Why is this not required here? We set the icon to the network icon previously.